chalice/cli/__init__.py

```diff
@@ -175,6 +175,23 @@ def deploy_new(ctx, autogen_policy, profile, api_gateway_stage, stage):
             config.project_dir, '.chalice', 'deployed.json'))
 
 
+@cli.command('delete-new')
+@click.option('--profile', help='Override profile at deploy time.')
+@click.option('--stage', default=DEFAULT_STAGE_NAME,
+              help='Name of the Chalice stage to delete.')
+@click.pass_context
+def delete_new(ctx, profile, stage):
+    # type: (click.Context, str, str) -> None
+    factory = ctx.obj['factory']  # type: CLIFactory
+    factory.profile = profile
+    config = factory.create_config_obj(chalice_stage_name=stage)
+    session = factory.create_botocore_session()
+    d = factory.create_deletion_deployer(session=session)
+    deployed_values = d.deploy(config, chalice_stage_name=stage)
+    record_deployed_values(deployed_values, os.path.join(
```

Comment:
Don't think that it needs to be updated in this PR, but ran into it when I was looking through the code, but the annotation for the `record_deployed_values` function looks off: https://github.com/jamesls/chalice/blob/79bd3b83624c80e2e29f51ccf3513c5ea5ca73d7/chalice/utils.py#L60. It is has the deployed values annotated as `Dict[str, str]` when it is really `Dict[str, Any]` now especially since the `deployed_values` contains each stage and a list of resources now.

Reply:
Interesting, I wonder why mypy didn't fail on this.  I'll update this.